src/AutoBuild.jl

```diff
@@ -140,10 +140,18 @@ supported ones. A few additional keyword arguments are accept:
   package. This can for example be used to invoke an initialization API of a
   shared library.
 
+* `augment_platform_block` may be set to a string containing Julia code; if
```

Comment:
Should we make this more general?  It isn't really specific to platform-augmentation, especially following https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/1128#discussion_r770987505.  I can see this argument can be (ab)used for other purposes as well.  What about `top_level_block` or something along this line?

Reply:
Scratch that, with new changes this is very specialised to platform-augmentation